app/src/main/java/com/example/nuberjam/ui/main/library/detail/DetailLibraryFragment.kt

```diff
@@ -52,11 +56,52 @@ class DetailLibraryFragment : Fragment() {
         super.onViewCreated(view, savedInstanceState)
 
         setupAppbar()
+        setupViewType()
         setupRecyclerView()
         showSnackbarObserve()
         initObserver()
     }
 
+    private fun setupAppbar() {
+        val toolbar: Toolbar = binding.appbar.toolbar
+        toolbar.navigationIcon = ContextCompat.getDrawable(
+            requireContext(), R.drawable.ic_back_gray
+        )
+        toolbar.setNavigationOnClickListener {
+            findNavController().popBackStack()
+        }
+        binding.appbar.btnSearch.setOnClickListener {
+            // TODO: navigate to search
+            Toast.makeText(requireActivity(), "You clicked me.", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private fun setupViewType() {
+        val args: DetailLibraryFragmentArgs by navArgs()
+        libraryViewType = args.viewType
+
+        with(binding) {
+            when (libraryViewType) {
+                LibraryDetailType.Favorite -> {
+                    appbar.tvLibraryAppbar.text = getString(R.string.liked_song)
+                    viewModel.getFavoriteData()
```

Comment:
don't call API data in fragment/activity, because it will refresh every fragment that appears. so call init API data in view model, then data will not refresh when fragment appears after pop another fragment.